src/vehicle.cpp

```diff
@@ -4762,6 +4882,9 @@ void vehicle::refresh()
     alternator_load = 0;
     camera_epower = 0;
     has_atomic_lights = false;
+    plow_engine_drag = 0;
```

Comment:
Instead of special casing them all the same way, add a flag like "POWERED_PART" abd trigger off of that.